### ISSUE TITLE
process on GPU

### ISSUE THREAD

**Issue by Doktor1024** [phrutis/LostCoins#51]
If the list is loaded into the GPU's internal memory, then please answer - how it happens literally. For example, I have 10 GPU 3060, each memory is 12 Gb, the list weighs 1Gb, the range is 2^100-2^101. How is the process of distributing the entire list of Sort 1 G.bin between all GPUs? if you say the list is loaded into the GPU, then how exactly will it load into 10 GPU units? 0.1Gb in parts for each GPU? If so, it is not clear how the comparison process proceeds throughout the list! It turns out that each GPU can check only its part of the list? Please explain - how does this happen at the level of the program and computer and GPU? If one GPU card or a search on processor cores - everything is clear --the whole list is in one place--PC RAM!

I'm not a programmer, I'm a mathematician! I want to understand how it works on 10 pieces of GPU? In my screenshots, everything is clear when working on the processor! But here's how the process will happen on 10 GPU units -- I don't understand! If the entire list is loaded into each card as a whole, then it turns out that it is the same as if it were 10 different people with 1 GPU!!! And if the parts from the list 1 \ 10 are loaded, then it means that each card is not looking for the entire list, but only for 1 \10 of the list! Please explain! I think your program is very good, but I don't understand how it works! 10 GPUs are not cheap))) I would like to understand the process! Sincerely, Doctor

https://mega.nz/file/QGAnwZZZ#dzYeJcsA75RsTeHTq8SNG_Yqx4XM0b-IrM-x7Z1QKJ0
https://mega.nz/file/peJjUDzR#kBnkJpvOoBXgGMCia_1OTwbtf34hvF9x50vsopt0jH0

**Comment by marssystems** [comment]
Intro to CUDA - An introduction, how-to, to NVIDIA's GPU parallel programming architecture - Bing video<https://www.bing.com/videos/search?q=nvidia+cuda&docid=607997554977612088&mid=2589AD23F5A5EAD1E4272589AD23F5A5EAD1E427&view=detail&FORM=VIRE>

[http://www.bing.com/sa/simg/facebook_sharing_2.png]<https://www.bing.com/videos/search?q=nvidia+cuda&docid=607997554977612088&mid=2589AD23F5A5EAD1E4272589AD23F5A5EAD1E427&view=detail&FORM=VIRE>
nvidia cuda - Bing<https://www.bing.com/videos/search?q=nvidia+cuda&docid=607997554977612088&mid=2589AD23F5A5EAD1E4272589AD23F5A5EAD1E427&view=detail&FORM=VIRE>
Intelligent search from Bing makes it easier to quickly find what you’re looking for and rewards you.
www.bing.com

________________________________
From: Doktor1024 ***@***.***>
Sent: 02 April 2022 21:18
To: phrutis/LostCoins ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [phrutis/LostCoins] process on GPU (Issue #51)


If the list is loaded into the GPU's internal memory, then please answer - how it happens literally. For example, I have 10 GPU 3060, each memory is 12 Gb, the list weighs 1Gb, the range is 2^100-2^101. How is the process of distributing the entire list of Sort 1 G.bin between all GPUs? if you say the list is loaded into the GPU, then how exactly will it load into 10 GPU units? 0.1Gb in parts for each GPU? If so, it is not clear how the comparison process proceeds throughout the list! It turns out that each GPU can check only its part of the list? Please explain - how does this happen at the level of the program and computer and GPU? If one GPU card or a search on processor cores - everything is clear --the whole list is in one place--PC RAM!

I'm not a programmer, I'm a mathematician! I want to understand how it works on 10 pieces of GPU? In my screenshots, everything is clear when working on the processor! But here's how the process will happen on 10 GPU units -- I don't understand! If the entire list is loaded into each card as a whole, then it turns out that it is the same as if it were 10 different people with 1 GPU!!! And if the parts from the list 1 \ 10 are loaded, then it means that each card is not looking for the entire list, but only for 1 \10 of the list! Please explain! I think your program is very good, but I don't understand how it works! 10 GPUs are not cheap))) I would like to understand the process! Sincerely, Doctor

https://mega.nz/file/QGAnwZZZ#dzYeJcsA75RsTeHTq8SNG_Yqx4XM0b-IrM-x7Z1QKJ0
https://mega.nz/file/peJjUDzR#kBnkJpvOoBXgGMCia_1OTwbtf34hvF9x50vsopt0jH0

**Comment by Doktor1024** [comment]
Thanks! As I understand it, all GPU cards are combined into one system of CUDA devices and they are all engaged in solving one task. Then I think it makes sense to introduce another function into the program - the exact determination of the number of CUDA devices (and not the forced writing of this number by the user)! You also need to enter the grid definition for each device 128x128 256x265..! Or write a separate program that determines which grid the GPU has! Well, or lay out a table with all known GPU models with information about their grids and limitations!

**Comment by marssystems** [comment]
If you omit the --gpux (-x) parameter, the exact number of grids will be assigned automatically by the program. No need for anything else - try it.

**Comment by Doktor1024** [comment]
Thanks, I'll try! As I understand it, I can't load a very heavy list (I have my own special lists))) which is bigger than the GPU memory? or what is the total memory of all the pieces of the GPU? or what is the mass of "blooming"? If I have 10 GPU 3060 12 GB RAM = 120 GB, how big a list can I upload! Have you looked at my list? in the screenshots above? It's small!)

**Comment by marssystems** [comment]
All I can say is - try it! If the list is too big to fit in memory you will get an error. Then you can reduce the size of the list to suit.

**Comment by Doktor1024** [comment]
Thanks!

**Comment by phrutis** [comment]
View closed questions.
The address list is loaded into each GPU.
When starting several GPUs, only the range is divided between the cards. Each card checks its segment against its complete database of all addresses. The GPU cannot load a base larger than the internal memory.
The program uses only addresses for 1 ... there are only 25 million of them.
It's only 1GB of memory. On 3090 24 GB there were tests of a large database with beacons. I don’t remember, but it seems with a base of more than 200 million. bloom filter starts to pass. Look in the answers.

**Comment by Doktor1024** [comment]
Thanks!

**Comment by Radar-7528** [comment]
Как сделать eth_base.bin 110 881 000 штук адресов??????

**Comment by Doktor1024** [comment]
Etherium I don't know! I can only make Bitcoin!

**Comment by Radar-7528** [comment]
> Эфириум не знаю! Я могу только сделать биткойн! почему 110 881 000 адресов?)) это магическое число? или это логика поиска? Прости
Complete base Eth+btc in rmd160

**Comment by Doktor1024** [comment]
put together two large address files .txt btc+eth , then do it .bin, then do the sorting! You don't know how to fold?

**Comment by marssystems** [comment]
How to create a base of Et[hereum addresses

Downloading the fresh database of ethereum addresses here  https://blockchair.com/dumps 
Use Python: python eth_addresses_to_bin.py eth_addresses_in.txt eth_addresses_out.bin
Next, you need to sort the base eth_addresses_out.bin
CMD: BinSort.exe 20 eth_addresses_out.bin eth_addresses_sotrted.bin
Or another program CMD: RMD160-Sort.exe eth_addresses_out.bin eth_addresses_sorted.bin

**Comment by Doktor1024** [comment]
In the command line (console), enter cmd /c copy E:\1.txt + E:\2.txt E:\3.txt you get the amount 3.txt

**Comment by Doktor1024** [comment]
I don't understand why you need two bases in one?

**Comment by marssystems** [comment]
You will find it in - https://github.com/phrutis/Rotor-Cuda/tree/main/Others

**Comment by Radar-7528** [comment]
> You will find it in - https://github.com/phrutis/Rotor-Cuda/tree/main/Others
eth_addresses_to_bin.py

**Comment by Doktor1024** [comment]
marrssistem,please read my topic "step" - https://github.com/phrutis/LostCoins/issues/52   is it possible to make a "step"?

**Comment by Doktor1024** [comment]
marrssistem, when you read it, do you understand my logic?

**Comment by marssystems** [comment]
You  have a list of 500,000,000 addresses - what is the point of that?

Most of those addresses will be empty - 0 BTC - so there is no point in searching for them and no need to generate such large lists in the manner you describe above.

All you need to do is prepare a list of approx. 50 addresses that have a large amount of lost BTC in them and search for those.

**Comment by Doktor1024** [comment]
for example, I have an address, I generate +500000000 addresses from it and search for any address from the list, and I found the private key to the address = 1000020! I take and look at what is the serial number of the address in the list 1,2,3.......500000000, and this number = 4444 in the list! 1000020 minus 4444 = private key of the first address!!!

**Comment by Doktor1024** [comment]
You are shooting at a lot of small targets!! I'm shooting at one fat\big target!

**Comment by marssystems** [comment]
Its easy to calculate the wallet address from any private key - its what your bitcoin wallet does when you ask for a new address.

Its impossible to go from address to private key because of the SECP256K1 algorithm so you have to create a wallet address from a private key (random or sequential) and see if it is on your list.

**Comment by Doktor1024** [comment]
you didn't understand anything again!!))

**Comment by Doktor1024** [comment]
1 adr   key x            this my target
2 adr   key x+1
3 adr   key x+2
4 adr   key x+3
5 adr   key x+4
6 adr   key x+5
7 adr   key x+6
**8 adr   key x+7**  this found key= fff000    I can find the key for 1st address))) fff000-7
9 adr   key x +8  
10......
11......
....
....
500000000 key+499999999

**Comment by Doktor1024** [comment]
understend??????

**Comment by Doktor1024** [comment]
if I know the public key - 02 03 04 coordinates (to the address) - I can create a list

**Comment by marssystems** [comment]
Yes - I understand that.

But how do you go from that, to finding the private key for address - 1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK - for example?

**Comment by Doktor1024** [comment]
if there was at least one payment from the address, then the public key (coordinates) is visible in the blockchain!!!

**Comment by marssystems** [comment]
Yes - I know that - but you can't calculate the private key from the public key.

**Comment by Doktor1024** [comment]
I can generate a large list of addresses- in this list, all addresses will be associated with the one I'm looking for! with the first one!! If I find any key from the list, I can calculate the key to the first one!!)) You wrote - you understand!!!

**Comment by Doktor1024** [comment]
There are 20,000 coins on your address and there are outgoing payments - you can create a list - of any length

**Comment by marssystems** [comment]
Yes - I know - and I can find the public key for that address. But there is no way to find the private key from that information.

**Comment by Doktor1024** [comment]
Probably there is a difference - either you shoot at one address 1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK - or you shoot at 500,000,000 addresses? what do you think?

**Comment by marssystems** [comment]
Most of the 500 million addresses will be empty. This is a waste of time and processing power.

I you have a list of sequential public keys that does not mean that the private keys will be sequential.

**Comment by Doktor1024** [comment]
You won't find it-and I know it!!! But when the target is 500000000 - it's easier to guess than one address! If you guess 1 out of 500000000, you will find your own one 
1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK

**Comment by Doktor1024** [comment]
))) I have the ability to create such a list - having only the public key! And in this list, private keys will go in a row!

**Comment by Doktor1024** [comment]
How?   I drew it at the very beginning!!

**Comment by Doktor1024** [comment]
I realized what moment you don't understand!!!

**Comment by marssystems** [comment]
No they won't. They are generated randomly - not sequentially. That is why the bitcoin addresses are so secure.

**Comment by marssystems** [comment]
A list of sequential public keys does not mean that the private keys will be sequential as well.

You are missing the point.

**Comment by Doktor1024** [comment]
))) Give me the address (you can)empty (but with a public key) and I will generate the following for 1 private above - list  so that the private one is known to you!! then you will check that other addresses in the list have +1 +2 +3 private

**Comment by marssystems** [comment]
Read this -
https://medium.com/coinmonks/how-to-generate-a-bitcoin-address-step-by-step-9d7fcbf1ad0b

**Comment by Doktor1024** [comment]
https://kriptokurs.ru/bitcointools/tool/pubkey-to-address   As you want))) or you can give the address))

**Comment by marssystems** [comment]
You can find the public key here - https://btc.com/btc/address/1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK

**Comment by Doktor1024** [comment]
Yes!! But you don't know his private! How can you check my list later? The fact that it contains addresses with a row +1+2+3 a private lot of your address? You give me the address with the public key, don't tell me about the private one - then you'll check it yourself by adding +1 to your private key!

**Comment by marssystems** [comment]
OK - this is the public key -

025ad9cc469e32ea3d294a152b78bfded6ad47cfb9d5d9e52e201cd93d0c05fae1

**Comment by Doktor1024** [comment]
Вот это хорошо!!! Через час

**Comment by marssystems** [comment]
OK

**Comment by Doktor1024** [comment]
I made a small one in order for you to believe me! Tomorrow I can explain in detail how it works!
1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK
1DPUuJtngzd9w96gJ6Z6ocLkFszVpSwCQ
1K3DzzReh2sN3Xo82c38KZTkc7wKmQLX82
1NFSRDNX2wt2imtZuB3T4mAC22CkU8Xyi5
1M8xZDaAc5c4BpMYyvwKaYgroQhHb45ma5
166anfp1k3XUkb72anfn8ZA8snF4mabaJv
1CR6biPi2bGQ8M7wr2kc2qSqyy3Bmwbyma
1GRMLgxRVctSd7ZQGhLnjRzef7iAVjwdGb
1L1GXbRV7DKhpudojZ2BZacbfsEHDJ9oaX
1GyKS2qkUMFQti6iFufn37TvfCosJ43pWQ
1JoLimeJ2V2hCxEREwbzyPsC4wmzwJPxhF
1JSDb9247rwfp3fLLdmR5sEXvs6QSBBFoR
15PRSUMfBEd6kXgqCzxePAX3EpYHhBtutk
1EvM3GnRPuiHe84kwJbtbsX9GpUBLTMCJH
1DPCmPaE9hEqWoneYQMkkwEgZsdYuGjruf

**Comment by marssystems** [comment]
OK - thanks!

**Comment by Doktor1024** [comment]
How is the weather in London?

**Comment by marssystems** [comment]
Cold and wet

**Comment by Doktor1024** [comment]
I just didn't understand!)) Why did you give me a public key to an address with so many coins in it? you don't know the private one from this address, do you? otherwise, why search with such a wealth of 10,000 coins? And if you don't know the private key to this address, then how can you check my little list? Give me another public key from which you know the private key. Although maybe you know the key to 10,000 coins!))

**Comment by Doktor1024** [comment]
You can generate a pair of private key to public and address (compress) here - https://secretscan.org/PrivateKeyHex Then give me the public key - don't tell me the private one, but keep it!

**Comment by marssystems** [comment]
OK - I picked a random private key and generated this public key -

03EE7A70A738652AF61CF50B59BC63904EB97421AF8C1DE1C2231202F7F1B30C23

See what you can do with that.

**Comment by Doktor1024** [comment]
Good morning!) Add +1 to your private key, don't forget HEX\DEC 16\10

1EUqLawaqhb8mFxnXgMiQ8VAhN4fpEswtG
1GXfWuhiQwJc6wmATD6oHVdCw7e8wZVcMb
15M3zDm8GEdbLPtA3YtCyUBbbTnyYVmrDh
16wf3NiQKAFHDm78PLT3dmmqFxx85v9nYM
1KcLZVrYGKPpVtr5USxDmUShya7DLqRDBe
1LvbEmrkfM7h5y3h8ES6BUUvjL14NQ3XgN
12AJz3aEpCvbwj5Edp893iBrswcky7f3uy
1KqZa7iFowePj7f38XoSWEkHXCbRPHXa3H
154uzSPi1aChCRbbs6qhMP7tAw3BBabruP
1BAvuYRmXgwcSga4d41NsD9GV1XkMUfTdA
1FrNBJnKCDX4jniCwz5KiMfPEs5gvj81Qw
1AqVtC6M2LCh2tJyUWcztGvxREfrEo8QtQ
1KQZDk9Dnpox5w1nPaU1rYANqufDdyX6wF
1FPKcY8KuZh24J6K6MszBoXcGq6AvqkxBh
1GsrnRvmk3wsQyAi9TDQQtvCKMT8J2LpX2
19FgkL5u8VqNbVVKFFfYfr2SC7VjFto8KB
16DXzvJ6c9Mb1KszJEcTjQZPyddLCxK8DE
1B1tB6tNYjSghxm9srkP7yVFiLXDg9CgNM
1C86P1U3T1fzeyo9Ya6wMFW6XT175czhoS
13BWefjLsDCvKwdcJdrpCjgXXcQWiLAfQJ
12aFjTzKomWB2zmo28X6rGxFj7QFigJfhp

**Comment by marssystems** [comment]
So what do I do now?
Do I search for these addresses using LostCoins.exe?

**Comment by Doktor1024** [comment]
Well, yes! I form this list with both software and hands! I will write in detail later how the list is formed! I think first of all it is necessary to convince the author of Lost Coins to enter the "step" function - then the search will be much faster and more chances! This list is small - I create 100000000-1000000000

**Comment by Doktor1024** [comment]
the author of Lost Coins - also don't not believe me!))) or doesn't understand!!! You write to him

**Comment by Doktor1024** [comment]
The size of the list is for example 1000 addresses! We are making a "step" of generating private keys - 999! When the program steps 999 - it will get to the size of 1000! Do you understand?

**Comment by marssystems** [comment]
Yes I do.

**Comment by marssystems** [comment]
Is this list you gave me a bit small? Don't I need a list of 500 million addresses to be effective in the search?

**Comment by Doktor1024** [comment]
What do you think is the difference -- there is 1 bingo ball in the lototron or there are 500,000,000 bingo balls in the lototron?

**Comment by marssystems** [comment]
I think the more addresses you have on the list, the higher the chance of finding the private key of one and then work backwards to get the key of the address you are interested in.

But I will now try with the list you gave me - maybe I will get lucky !!

**Comment by Doktor1024** [comment]
)))  OK! I think he's a very small list! What mode do you want to search for? in Lost Coins. 4 method (random bit range)? I can create you a bigger list -- it's free for patience and understanding!))) I can do it in the evening! Specify for which public? where are 10,000 coins?

**Comment by marssystems** [comment]
Yes - mode 4 is the best.

**Comment by Doktor1024** [comment]
I see that you answer here very often in tickets! I thought you were on the same team as the author of Lost Coins! well, or friends! Please write to him about my logic and my lists - let him make a "step" function! You are welcome! Maybe I have bad English and he doesn't understand me!? It's not me - it's Google Translator!)))

**Comment by marssystems** [comment]
Yes you can do a list for - 1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK - (10,000 BTC)

025ad9cc469e32ea3d294a152b78bfded6ad47cfb9d5d9e52e201cd93d0c05fae1

**Comment by Doktor1024** [comment]
What are your capacities - GPU - internal memory of the card? or a lot of GPU? What size list should I make?

**Comment by Doktor1024** [comment]
100000000 адрессов - .....2Gb( its bin sort file)

**Comment by Doktor1024** [comment]
Do you need to make a list of addresses or a ready-made sort.bin file?

**Comment by marssystems** [comment]
I have 5 X  3060 Ti 8Gb - 32Gb system memory.

Probably 25 million addresses will be fine.

**Comment by Doktor1024** [comment]
ок

**Comment by marssystems** [comment]
No - I am not on the programming team. I am just an interested user like yourself.

**Comment by Doktor1024** [comment]
Well, anyway - write to him in good English! Please, I think he read my tickets and he thought I was crazy or ignorant or didn't know elliptic curves!))) I'm a mathematician! But not a programmer

**Comment by Doktor1024** [comment]
While I'm generating a large list, I suggest playing with a small list! I generated 1000 addresses from the private key (fff), the list also contains 1000 addresses opposite (replacement 02-03 03-02)! You can choose a range in the 4 Lostcoins method, put a list with one first address 1Lv5xBNt4TYcf9DqF73WJAhzB8mG1cZH8P for the search first, then try with a list of 2002 addresses! You have a lot of GPUs - give a big range! And compare - how much time he spends on one address, and then how much time on the list! file - Sortw02+w03.bin

https://mega.nz/folder/cH4h3KKY#HVRphshYQtXlcbfGl7imGw

**Comment by marssystems** [comment]
OK - I have copied the software - you can delete the link - thanks.

**Comment by Doktor1024** [comment]
I have generated 30 million addresses! The list is made this way - you can search in the first half of all private keys (or in the second) - LostCoins checks addresses from 02 03 of one public key! But don't look for all private keys on the entire range - it's 2 times more! or 2^2 -2^254 or 2^254-2^256

https://mega.nz/file/JDgRQCTA#2aII5RdO5BNZ8I03moI__0LSGCpPczqTtYX4OKGRdt0

**Comment by Doktor1024** [comment]
sorry  its link https://mega.nz/file/BSBm0AyR#-eiie4fThXtUNkOBVRaWeDyend70vfxlxxdxT47WmME

**Comment by marssystems** [comment]
OK - thanks.

**Comment by marssystems** [comment]
When I find the private key to one of the 30 million addresses how do I calculate the private key of - 1LfV1tSt3KNyHpFJnAzrqsLFdeD2EvU1MK - the wallet address that I am interested in?

**Comment by Doktor1024** [comment]
)))if such happiness happens, then I think it will be fair 50\50? what do you think? my magic list is your GPU!! I think there are many options - 1) you give me the found key and your wallet, after 12 hours you have 5000- 12 hours - I need to generate a list of addresses again-I can't store them - my disk is not a risin! I generate, find, count and happiness for both you and me! 2) Skype conference - I am open to the whole world!)) 3) of meeting in a good restaurant!))and we do everything eye to eye!

**Comment by Doktor1024** [comment]
I first create a list of addresses, then a bin file, then sort, then gave you the sorting! I can't keep all the address lists! But this list is easily generated again! точно такой же!

**Comment by Doktor1024** [comment]
I think it's fair

**Comment by Doktor1024** [comment]
If you doubt the list - do not doubt!!! This is a real list! If you find one key, you will find 10,000!!!

**Comment by marssystems** [comment]
OK

**Comment by Doktor1024** [comment]
ok

**Comment by Doktor1024** [comment]
If you want, here are two more lists! This is for a puzzle of 119-120 bits, 1.2 coins! The first list is 101 million addresses, the second 500 million addresses, but I'm not sure it will fit in memory! The chances are higher - the range is small! Mode 4, range 119 -120 bits

https://mega.nz/file/RLgQ0DLL#eg7wgBgEkVnl0qmnI5pQxFPDIHI8h58FVqPVaS18ybo

https://mega.nz/file/cTQDXYoA#tKrzAEiWK69RvGopIGQZE-og9Id4xQL7vxx9BeZroys

**Comment by marssystems** [comment]
I'm currently running an AI simulation for another project. It should be finished tomorrow afternoon.
I will try these files then.

**Comment by Doktor1024** [comment]
ок 119-120 bit 4 mode

**Comment by marssystems** [comment]
Yes

**Comment by Doktor1024** [comment]
I am sure that all the addresses are in the 500M and 150M lists - since the edges of the range 119 + and 120- were checked for such a number of addresses !!! The 101M list is generated with a small step +, 1 million for each step, then these steps are added together! There is some chance that some steps fly out from the side of the big edge of 120 bits!

**Comment by Doktor1024** [comment]
150 and 500 - go in a row +1 from the address point 120pazl

**Comment by Doktor1024** [comment]
101M goes from the point 1M + step and again 1M + step and again 1M ...... only 100 steps and 1 mod of the point itself in a row +1

**Comment by Doktor1024** [comment]
Please tell me what the string looks like (if it is in LostCoins) to search randomly from a range (not a bit) - for example (fffff-ffffcccccccc) to the private_key range? Random - not sequential!

**Comment by marssystems** [comment]
You can't use  --range fffff-ffffcccccccc in LostCoins . You have to use -s  and -z  to specify the range.

**Comment by marssystems** [comment]
Use Rotor-Cuda if you want to specify a range.

**Comment by Doktor1024** [comment]
I know it!))) I mean, what does the string look like and what mode is it? Range and mode -what does the string look like? random is not consistent! Please write a line! Mode 4 is random bits! 1 mode is sequential! I need random - not bitwise! in method 4, there are random bits, and I need the range fffffff-ffcccddd

**Comment by Doktor1024** [comment]
I understand there is no such mode! Conditions - 1) compare with the list! 2) range (not bits) ff-fffcccddd111 3) random within the range! In the Rotor - there is no comparison with the list, only 1 adress

**Comment by Doktor1024** [comment]
I want to divide 2^119 - 2^120 - manually into small ranges - 800000000000000000000000000001 - 800000000000000fff00000000000001 is an example

**Comment by Doktor1024** [comment]
You anderstand?

**Comment by marssystems** [comment]
You can do that in Rotor-Cuda.

**Comment by marssystems** [comment]
https://github.com/phrutis/Rotor-Cuda

**Comment by Doktor1024** [comment]
I can't compare with the list (many addresses) in the Rotor!!!  Or am I wrong? I think the Rotor works with one address

**Comment by marssystems** [comment]
Yes you can.

Rotor works the same as LostCoins but has more options and is faster.

**Comment by Doktor1024** [comment]
Please write a command line for the Rotor! The range is ffff-fffff . File 101M.bin . Random search in the ffff-ffffff range

**Comment by marssystems** [comment]
Rotor-Cuda.exe -g --gpui 0 --gpux 256,256 -m address --coin BTC -r 100 --range ffff:ffffff -i 101M.bin

**Comment by Doktor1024** [comment]
Thanks

**Comment by marssystems** [comment]
I've been running the Sort150M.bin file on the 120 bit puzzle for 10 days now and I haven't found anything yet.
Are you sure that the 150 million addresses are in the 120 bit range?

**Comment by Doktor1024** [comment]
Yes, I'm sure!!! I took a point and added +1 to it! What range are you looking for - 2^119 2^120?

**Comment by Doktor1024** [comment]
If you want, I can give you the software! You'll make your own lists! Only you have to understand - only I and the programmer have this software - no one else knows about it! The programmer himself doesn't really understand why either!! I can give for free! You have to understand - the more people know, the less our chances are!

**Comment by marssystems** [comment]
This range is not that big so I'm a bit confused as to why I haven't found anything after 10 days.

**Comment by marssystems** [comment]
Your software sounds interesting. I wouldn't mind producing some 'test' files to prove that your hypothesis is correct.

**Comment by Doktor1024** [comment]
https://mega.nz/file/lbpizAaR#K2hLclNnBt9CPG-dLIKNcfsf7q_IB0sOkdK-mMAdewk

https://mega.nz/file/NPYB0TDJ#qUljcXGxwmHzzCPQlrarkT-CLYrmoVNXpBhtIf-dxM4


Take it away - I'll delete it! You have an illusion - it's still a very large range!

**Comment by Doktor1024** [comment]
I'll explain how it works!

**Comment by marssystems** [comment]
Yes - some instructions will be good.

**Comment by Doktor1024** [comment]
There are two programs! 1 -"main" ! You launch and specify the coordinates of X Y (public key), press enter, enter the number of additions (1 million......), the program asks - Save X and Y or save only X -- you choose - only X - dial 01 - press enter

**Comment by marssystems** [comment]
Ok - I understand that.

**Comment by Doktor1024** [comment]
This is the first stage! You get a list X +1+1+1+1+1...... from the original public key!

**Comment by marssystems** [comment]
Is this a list of public keys?

**Comment by Doktor1024** [comment]
The second program asks you - the name of the file with X coordinates! Everything should happen in one folder! You specify the file name X coordinates! The program asks which prefix to generate -02 or 03, you choose 02 first - make a list, then 03 -make a list, then add them up! The resulting list will contain 50% of the addresses you need! And 50% of the addresses "e.g. " G - " - with the same X

**Comment by marssystems** [comment]
OK

**Comment by marssystems** [comment]
Is the output file from the first stage generated automatically or do I have to give it as a parameter to the 'main' program.

**Comment by Doktor1024** [comment]
You get a list with X coordinate! In the second program, you specify the name of this list and choose 02 or 03! To get a list of addresses for LostCoins - you need only one prefix - either 02 or 03! LostCoins checks two prefixes at once and 02 and 03! Having one - you will shoot at the target!  The rotor -program -checks for only one prefix!

**Comment by marssystems** [comment]
OK

**Comment by Doktor1024** [comment]
https://mega.nz/file/oT5ngTrY#f9iGOUr9Jv7bkamdnRbKxeNsfH2Ouq6T4hwk392A9jo

For control and help!)))

**Comment by Doktor1024** [comment]
The second generates "compressed addresses"

**Comment by Doktor1024** [comment]
After generating the list of addresses , you need to translate to .bin and sort!

**Comment by marssystems** [comment]
OK - I understand

**Comment by Doktor1024** [comment]
Ask if the nuances are unclear! Good luck! If - FOUND - I HOPE YOU WON'T FORGET ME!)

**Comment by marssystems** [comment]
Yes - if I have any problems, I will ask you.

**Comment by Doktor1024** [comment]
Ok! I'll give you a picture tomorrow! I will draw on it in detail and clearly over the entire range of 2^256

**Comment by marssystems** [comment]
How many GPUs do you have in your system?

**Comment by Doktor1024** [comment]
3 3060 12GB

**Comment by marssystems** [comment]
OK

**Comment by Doktor1024** [comment]
If you want to create a list of addresses for LostCoins, it's enough to run one prefix 02 or 03! If for Rotor - it is necessary to generate 2 lists - 02 list and 03 list, after adding these two lists - you will get a list in which 50% of the necessary addresses and 50 opposite (the same X)! It doesn't matter how to fold - then they all move - sorting bits

**Comment by marssystems** [comment]
I will try over the weekend

**Comment by Doktor1024** [comment]
Try this list in the range 2^10 2^113 

https://mega.nz/file/APQjhShZ#SA16Zvbdt5r91jEbsiv0S8yfczG7S6nQ-ObSyCZLr-o

this is a 120 puzzle brought to the beginning

**Comment by Doktor1024** [comment]
or 64 - 147AE147AE147AE147AE147AE147A in Rotor
